core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1374,10 +1387,17 @@ private Optional<RetentionSizeData> buildRetentionSizeData(long retentionSize,
                     Iterator<RemoteLogSegmentMetadata> segmentsIterator = remoteLogMetadataManager.listRemoteLogSegments(topicIdPartition, epoch);
                     while (segmentsIterator.hasNext()) {
                         RemoteLogSegmentMetadata segmentMetadata = segmentsIterator.next();
-                        RemoteLogSegmentId segmentId = segmentMetadata.remoteLogSegmentId();
-                        if (!visitedSegmentIds.contains(segmentId) && isRemoteSegmentWithinLeaderEpochs(segmentMetadata, logEndOffset, epochEntries)) {
-                            remoteLogSizeBytes += segmentMetadata.segmentSizeInBytes();
-                            visitedSegmentIds.add(segmentId);
+                        // Only count the size of "COPY_SEGMENT_FINISHED" and "DELETE_SEGMENT_STARTED" state segments
+                        // because "COPY_SEGMENT_STARED" means copy didn't complete, and "DELETE_SEGMENT_FINISHED" means delete completed.
+                        // Note: there might be some "COPY_SEGMENT_STARED" segments not counted here, but become "COPY_SEGMENT_FINISHED" soon.
+                        // It's fine because the missed segment size will be count in next time, and it won't cause more segment deletion.
```

Comment:
```suggestion
                        // Only count the size of "COPY_SEGMENT_FINISHED" and "DELETE_SEGMENT_STARTED" state segments
                        // because "COPY_SEGMENT_STARTED" means copy didn't complete, and "DELETE_SEGMENT_FINISHED" means delete did complete.
                        // Note: there might be some "COPY_SEGMENT_STARTED" segments not counted here.
                        // Either they are being copied and will be counted next time or they are dangling and will be cleaned 
                        // elsewhere.
                        // Either way, this won't cause more segment deletion.
```